docs/docs/dynamo-store/access-strategies.md

```diff
@@ -70,3 +71,63 @@ With this in place Equinox will maintain and store a Snapshot event in the tip
 document that is guaranteed to always be up to date, and as a result the state
 of the Stream can be reconstructed with a single point-read.
 
+## RollingState(toSnapshot)
+
+`RollingState` will throw away events, only storing the snapshot. This can be
+useful for cases where we do not care about the individual events that led up to
+a state. This is frequently used to build up a read model that spans streams.
+
+<details>
+<summary>Example</summary>
+
+```ts
+namespace Events {
+  export type Event = { type: "Ingested"; data: { version: number; users: string[] } }
+  export const codec = Codec.json<Event>()
+}
+
+namespace Fold {
+  export type State = { version: number; users: Set<string> }
+  export const initial = { version: 0, users: [] }
+  export const fold = (state: State, events: Events.Event[]): State => events[0].data
+  export const toSnapshot = (state: State): Events.Event => ({
+    type: "Ingested",
+    data: {
+      version: state.version,
+      users: Array.from(state.users),
+    },
+  })
+}
+
+namespace Decide {
+  export const addUser = (id: string) => (state: State) => {
+    if (state.users.has(id)) return []
+    const newUsers = new Set(state.users)
+    newUsers.add(id)
+    return [{ version: state.version + 1, users: newUsers }]
```

Comment:
hm, does not manage deletion so not sure this is an ideal example 
if it did have to, then it would have to keep versions of each input stream

Reply:
What is this delete you speak of? 😜 

Do you have an idea for a better simple example?

Reply:
good question
maybe track renames; that'll require keeping the ids and versions alongside the names

perhaps just keeping a map of ids to names and letting them cycle through all the superseded states in the context of handing a replay is not the end of the world

but I'd argue that you want to illustrate semi-interesting logic that you might want to test and/or not use native DDB update functions for

Reply:
Updated to handle deletes

Reply:
ok, now the renames
... and  prevent it revisiting old states under replay!

Reply:
Renames and all that landed

Reply:
This all probably needs to be caveated as well. If you have 100K items in the map this'll just blow up. You could solve for it by linking to a `next` stream but even then, for a large enough dataset it should just become its own table.

(I'm guessing you're gonna school me about [epochs](https://github.com/jet/dotnet-templates/blob/master/equinox-patterns/README.md#list-epochsseries-with-exactly-once-ingestion-guarantee))

Reply:
Updated is prob a better event type name now
L113 maybe you still want to update the version?
maybe add a createMem
maybe prepend a `$` to cat name

Reply:
the caveat is implied in the 400K limit
no I can't think of a reasonable way to do it with epochs as the point of them is that they are append only and you walk them so they are fine to grow
for an index like this, you need to be able to reasonably
a) load into memory and do a manipulation in same
b) be able to serialize and transport the updated set
c) pay for an upateitem call without it getting daft

So if you have lots of items all getting updated the contention would get massive, and RU costs to update a 200K item to twiddle a few bytes get prohibitive (they're not that much better if you do a custom schema and use sets - its down to how big the item is at the end of the day, even if you can express the desired change as a tiny update command)

To deal with that contention, you might also use a Batcher if you had concurrency in play

you'd need to hash across a fixed number of docs to partition it, and then apply the update logic to a smaller subset

but then to load you'd need to either do a cross partition scan, or e.g. do N cached point reads (which is not as insane as it sounds if N is low)

the `eqx query` command I am due to add soon would provide an API to query across things
so e.g. on cosmos it would do a cross partitiion query for tips of things
and then can render a response direct from that
in the case of a deletable renameable item set that's within a reasonably limited set of items where the items have snapshots and/or all events are likely in tip, you'd then query the states and render as a list rather than go to the trouble of keeping an index
and you could also render that list as a paged result set with continuation tokens

Maybe you also should consider illustrating how you'd apply a consecutive set of upstream events batched into an array (either for one thing if you had a propulsion doing batches per upstream thing, or N events across M streams if you had that and a batcher to coordinate concurrent handings such as that). But that's probably another article

Reply:
> the caveat is implied in the 400K limit

Glad we have tracing from which we can trigger alarms when a something like this grows to let's say 350K 😉 

> but then to load you'd need to either do a cross partition scan, or e.g. do N cached point reads (which is not as insane as it sounds if N is low)

I'd definitely do the N reads as a default.

> in the case of a deletable renameable item set that's within a reasonably limited set of items where the items have snapshots and/or all events are likely in tip, you'd then query the states and render as a list rather than go to the trouble of keeping an index

Where do you get the set of IDs to query?

Reply:
>Where do you get the set of IDs to query?

set of ids to query is a prefix or pattern query based on the streamname - i.e. p startswith and i=-1
if you knew the ids, that's an async.parallelLimit or a BatchGetItems